### PR TITLE
ESRF SSX queue entries, PyISPyB client and bug fixes.

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
This PR mainly consists of three parts;

 - Additions and updates to the ESRF SSX queue entries
 - PyISPyB client updates, the previous client in `utils` was moved to an external project (package on pypi) and the PyISPyB HardwareObject updated
 - One or two minor bug fixes and/or general improvements

### CODE REVIEW

File: mxcubecore/HardwareObjects/Session.py [hunk 1]
```diff
@@ -284,10 +284,10 @@ def get_default_subdir(self, sample_data: dict) -> str:
             sample_name = sample_data.name
             protein_acronym = sample_data.crystals[0].protein_acronym
 
-        if protein_acronym:
-            subdir = "%s/%s-%s/" % (protein_acronym, protein_acronym, sample_name)
-        else:
-            subdir = "%s/" % sample_name
+        #if protein_acronym:
```
Comment (meguiraun) [inline mxcubecore/HardwareObjects/Session.py]: why this change?

File: mxcubecore/utils/nicoproc.py [hunk 2]
```diff
@@ -2,8 +2,7 @@
 import xmlrpc.client
 
 USE_NICOPROC = False
-HOST = "http://localhost:9998/"
-
+HOST = "http://lid29control-2:9998"
```
Comment (meguiraun) [inline mxcubecore/utils/nicoproc.py]: would not this be esrf specific? so maybe better inside ESRF folder?

**Review by meguiraun — APPROVED** [review]
very small comments. looks good to me

File: mxcubecore/HardwareObjects/Session.py (hunk 1 above)
Comment (marcus-oscarsson) [reply mxcubecore/HardwareObjects/Session.py]: Ah, sorry that change should not be there. It was moved to `ESRFSession`  thanks for pointing it out !

File: mxcubecore/utils/nicoproc.py (hunk 2 above)
Comment (marcus-oscarsson) [reply mxcubecore/utils/nicoproc.py]: Yes indeed, Ill move that to a more ESRF specific location :)

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+
+__credits__ = ["MXCuBE collaboration"]
+__license__ = "LGPLv3+"
+__category__ = "General"
+
+
+class ExpTimeEnum(float, enum.Enum):
+    hundred_mirco = 100e-6
+    fifity_micro = 50e-6
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py]: Typo?

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class SSXUserCollectionParameters(BaseModel):
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py:33]: I am curious how this works - it looks like a model, so why is it in a queue_entry file?

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py [hunk 5]
```diff
@@ -68,112 +59,72 @@ class InjectorColletionTaskParameters(BaseModel):
 
 
 class SsxInjectorCollectionQueueModel(DataCollection):
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py:53]: Why the super()? It is a no-op, effectively. In fact one could have done
SsxInjectorCollectionQueueModel = DataCollection

File: mxcubecore/HardwareObjects/ESRF/queue_entry/test_collection.py [hunk 6]
```diff
@@ -59,13 +67,42 @@ class TestCollectionQueueEntry(BaseQueueEntry):
 
     # New style queue entry does not take view argument,
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/ESRF/queue_entry/test_collection.py:68]: Out-of-date comment?

File: mxcubecore/HardwareObjects/LimaJungfrauDetector.py [hunk 7]
```diff
@@ -88,7 +89,8 @@ def init(self):
                 "SetImageHeader",
             )
 
-            self.get_command_object("prepare_acq").set_device_timeout(30000)
+            # prepareAcq can block after large data transfers (buffered by GPFS): 2 min timeout
+            self.get_command_object("prepare_acq").set_device_timeout(2 * 60 * 1000)
```
Comment (rhfogh) [inline mxcubecore/HardwareObjects/LimaJungfrauDetector.py:93]: Normally we use seconds for time values and timeouts. Why miliseconds here? And, more important, where is that documented?

**Review by rhfogh — APPROVED** [review]
As you see I have a few questions out of curiosity, but nothing that i would insist on should hold you up.

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py (hunk 4 above)
Comment (marcus-oscarsson) [reply mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py:33]: Its to make the queue_entry somewhat self contained the models that are only used from a queue entry is defined in the queue entry file. It simply makes it easier to see what they are when one need to change or reference them.

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py (hunk 5 above)
Comment (marcus-oscarsson) [reply mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py:53]: No reason more than its then easier to extend it later if needed and `SsxInjectorCollectionQueueModel = DataCollection` looks a bit odd in mu opinion

File: mxcubecore/HardwareObjects/LimaJungfrauDetector.py (hunk 7 above)
Comment (marcus-oscarsson) [reply mxcubecore/HardwareObjects/LimaJungfrauDetector.py:93]: Its a call to a tango device and a Tango takes timeouts in milliseconds.

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py (hunk 4 above)
Comment (rhfogh) [reply mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_foil_collection_lima2.py:33]: Makes sense (when, as here, it is site-specific)

File: mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py (hunk 5 above)
Comment (rhfogh) [reply mxcubecore/HardwareObjects/ESRF/queue_entry/ssx_injector_collection.py:53]: The super() is still unnecessary - but OK, it eats no bread.

**Comment by marcus-oscarsson** [comment]
Thanks for the input :), Ill have a look :+1:

File: mxcubecore/HardwareObjects/LimaJungfrauDetector.py (hunk 7 above)
Comment (rhfogh) [reply mxcubecore/HardwareObjects/LimaJungfrauDetector.py:93]: Perfectly correct, then

**Comment by marcus-oscarsson** [comment]
Looks like there are still some issues with starting the CI container. The tests run fine locally so I think that part is fine. Lets see if GitHub manages to solve this issue (or if we need to update some dependency somewhere, perhaps in the conda-environment)
